provider/server/machine_create.go

```diff
@@ -78,6 +79,8 @@ func (s *Server) createMachineFromIRIMachine(ctx context.Context, log logr.Logge
 		return nil, fmt.Errorf("failed to set metadata: %w", err)
 	}
 	apiutils.SetClassLabel(machine, iriMachine.Spec.Class)
+	apiutils.SetMachineNamespaceLabel(machine, iriMachine.Metadata.Labels[ironcorev1alpha1.MachineNamespaceLabel])
+	apiutils.SetMachineNameLabel(machine, iriMachine.Metadata.Labels[ironcorev1alpha1.MachineNameLabel])
```

Comment:
If either `iriMachine.Metadata` or `iriMachine.Metadata.Labels` is empty we run in an nil-pointer exception

Reply:
can we solve it somewhere else? because we will have problem with it always. It can be solve as setter for metedata type which can handle this nil references.

Reply:
pls remove since the metadata is already stored here: 
```
	if err := apiutils.SetObjectMetadata(machine, iriMachine.Metadata); err != nil {
		return nil, fmt.Errorf("failed to set metadata: %w", err)
	}
```

Reply:
I acknowledge that if `iriMachine.Metadata` is empty, it could lead to a nil-pointer exception since Metadata is a pointer to `v1alpha1.ObjectMetadata`. To address this, I will include a condition to ensure that Metadata is not nil before proceeding.

On the other hand, `iriMachine.Metadata.Labels` is of type `map[string]string` and will be initialized to its zero value if left empty. Therefore, there is no need to be concerned about nil values. I have tested this scenario, and in such cases, the metadata in domainXML appears as follows:
```
<metadata>
    <libvirtprovider:metadata xmlns:libvirtprovider="https://github.com/ironcore-dev/libvirt-provider">
        <libvirtprovider:namespace/>
        <libvirtprovider:name/>
    </libvirtprovider:metadata>
</metadata>
```
This output demonstrates that even when`iriMachine.Metadata.Labels` is empty, the XML representation is well-formed and contains the expected structure.

Please let me know if, this will address your concern.

Reply:
I would propose to store the data only once and fetch it in the controller directly at the correct place.

Reply:
ok I got your point, we can extract it from the value of annotation `libvirt-provider.ironcore.dev/labels`. Correct?

However, still your concern remains that if  `iriMachine.Metadata` is empty, then `if err := apiutils.SetObjectMetadata(machine, iriMachine.Metadata); err != nil {` will lead to nil-pointer exception.
Also in some places before using `iriMachine.Spec` the nil pointer is not bein checked, even that can be problamatic.

Reply:
@so-sahu be careful. map is reference type and zero value for map is nil: https://go.dev/tour/moretypes/19

Reply:
@lukas016  In Go, a nil map is a valid value for a map type, and you can perform read operations on it without causing a panic. If you try to access a key in a nil map, the result will be the zero value for the map's value type.
So, here, it's not a matter of concern.

Reply:
yes, you have true but write is problematic so pls be careful with zero values.